infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -219,11 +222,28 @@ static ByteBuffer getByteBufferWithMetaData(AbstractMessage message) {
     }
 
     @Override
-    public TailsResponse getTails() {
+    public StreamsAddressResponse getStreamsAddressSpace() {
+        Map<UUID, StreamAddressSpace> streamsAddressSpace = new HashMap<>(logMetadata.getStreamsAddressSpaceMap());
```

Comment:
isn't it a big overhead - providing an entire map? How often it is used?

Reply:
It is used on Sequencer Reconfiguration, we need the complete map to recover the sequencer.